ansible_ai_connect/ai/api/model_pipelines/llamastack/pipelines.py

```diff
@@ -97,7 +113,113 @@ class LlamaStackMetaData(MetaData[LlamaStackConfiguration]):
 
     def __init__(self, config: LlamaStackConfiguration):
         super().__init__(config=config)
+        self.headers = {"Content-Type": "application/json"}
+        i = self.config.timeout
+        self._timeout = int(i) if i is not None else None
+
+    def timeout(self, task_count=1):
+        return self._timeout * task_count if self._timeout else None
+
+    def self_test(self) -> HealthCheckSummary:
+        """
+        Perform a self-test to check the health of the LlamaStack provider.
+        Returns:
+            HealthCheckSummary: A summary of the health check results.
+        """
+        summary: HealthCheckSummary = HealthCheckSummary(
+            {
+                MODEL_MESH_HEALTH_CHECK_PROVIDER: "llama-stack",
+                MODEL_MESH_HEALTH_CHECK_MODELS: "ok",
+            }
+        )
+        try:
+            r = requests.get(
+                self.config.inference_url + "/v1/providers",
+                headers=self.headers,
+                timeout=self._timeout
+            )
+            # Check if the request was successful
+            r.raise_for_status()
+
+            data = r.json()
+            providers_data = data.get("data", [])
+            # Look for the vllm-inference provider
+            provider_found = False
+            for provider in providers_data:
+                if provider.get(LLAMA_STACK_PROVIDER_HEALTH) == LLAMA_STACK_PROVIDER_ID:
+                    provider_found = True
+                    health_status = provider.get("health", {}).get("status")
+                    if health_status != "OK":
+                        reason = f"Provider {LLAMA_STACK_PROVIDER_ID} health status: {health_status}"
+                        summary.add_exception(
+                            MODEL_MESH_HEALTH_CHECK_MODELS,
+                            HealthCheckSummaryException(ServiceUnavailable(reason)),
+                        )
+                    break
+            if not provider_found:
+                reason = f"Provider {LLAMA_STACK_PROVIDER_ID} not found"
+                summary.add_exception(
+                    MODEL_MESH_HEALTH_CHECK_MODELS,
+                    HealthCheckSummaryException(ServiceUnavailable(reason)),
+                )
 
+        except Exception as e:
+            logger.exception(str(e))
+            summary.add_exception(
+                MODEL_MESH_HEALTH_CHECK_MODELS,
+                HealthCheckSummaryException(ServiceUnavailable(str(e)), e),
+            )
+
+        return summary
+
+@Register(api_type="llama-stack")
+class LlamaStackChatBotPipeline(LlamaStackMetaData, ModelPipelineChatBot[LlamaStackConfiguration]):
```

Comment:
Added implementation for `LlamaStackChatBotPipeline`, as it was needed to show the status of `chatbot-service`, I understand that we might use just the streaming response from llama-stack provider end. So, if required I can remove this implementation